scipy/spatial/transform/rotation.py

```diff
@@ -1460,3 +1460,127 @@ def __getitem__(self, indexer):
+    rotations : `Rotation` instance
+        Rotations to perform the interpolation between. Must contain N
+        rotations.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
+
+    Examples
+    --------
+    >>> from scipy.spatial.transform import Rotation as R
+    >>> from scipy.spatial.transform import Slerp as S
+
+    Setup the fixed keyframe rotations and times:
+
+    >>> key_rots = R.from_quat(np.random.uniform(size=(5, 4)))
+    >>> key_times = [0, 1, 2, 3, 4]
+
+    Create the interpolator object:
+
+    >>> interpolator = S(key_times, key_rots)
+
+    Interpolate the rotations at the given times:
+
+    >>> times = [0, 0.5, 0.25, 1, 1.5, 2, 2.75, 3, 3.25, 3.60, 4]
+    >>> interp_rots = interpolator(times)
+
+    The interpolator returns the same rotations at the keyframe times:
+
+    >>> key_rots.as_quat()
+    array([[0.33546361, 0.32248502, 0.6144695 , 0.63709875],
+           [0.32914502, 0.13316173, 0.69664424, 0.62339258],
+           [0.71951125, 0.09194507, 0.28850306, 0.62499252],
+           [0.2432484 , 0.7620946 , 0.5988079 , 0.03835539],
+           [0.47888381, 0.47046693, 0.71809073, 0.18351259]])
+    >>> interp_rots.as_quat()
+    array([[0.33546361, 0.32248502, 0.6144695 , 0.63709875],
+           [0.33409754, 0.22905279, 0.65909448, 0.63364669],
+           [0.33523071, 0.27613969, 0.63763817, 0.63622701],
+           [0.32914502, 0.13316173, 0.69664424, 0.62339258],
+           [0.54669937, 0.11735566, 0.51359003, 0.65082465],
+           [0.71951125, 0.09194507, 0.28850306, 0.62499252],
+           [0.42131035, 0.65782128, 0.58309954, 0.22307772],
+           [0.2432484 , 0.7620946 , 0.5988079 , 0.03835539],
+           [0.30797834, 0.70009544, 0.63967903, 0.07633122],
+           [0.39267851, 0.60035814, 0.68483292, 0.12797487],
+           [0.47888381, 0.47046693, 0.71809073, 0.18351259]])
+    """
+    def __init__(self, times, rotations):
+        if len(rotations) == 1:
+            raise ValueError("`rotations` must contain at least 2 rotations.")
+
+        times = np.asarray(times)
+        if times.ndim != 1:
+            raise ValueError("Expected times to be specified in a 1 "
+                             "dimensional array, got {} "
+                             "dimensions.".format(times.ndim))
+
+        if times.shape[0] != len(rotations):
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of timestamps given, got {} rotations "
+                             "and {} timestamps.".format(
+                                len(rotations), times.shape[0]))
+        self.times = times
+        self.timedelta = np.diff(times)
+
+        if np.any(self.timedelta <= 0):
+            raise ValueError("Times must be in strictly increasing order.")
+
+        # inv() and __getitem__ return new objects anyway
```

Comment:
This comment is unclear without a context. You can remove it or explain better.

Reply:
I'll remove it. Basically, traditional array slicing returns a view on the object. Here that is not the case so we can modify the returned object as we like. However, that was quite a few commits ago and now we basically ensure that our object cannot be modified once initialized so this is no longer relevant.